src/backend/distributed/commands/role.c

```diff
@@ -141,6 +239,100 @@ ExtractEncryptedPassword(Oid roleOid)
 }
 
 
+/*
+ * GenerateAlterRoleSetIfExistsCommandList generate a list of ALTER ROLE .. SET commands that
+ * copies a role session defaults from the pg_db_role_settings table.
+ */
+static List *
+GenerateAlterRoleSetIfExistsCommandList(HeapTuple tuple, TupleDesc
+										DbRoleSettingDescription)
+{
+	AlterRoleSetStmt *stmt = makeNode(AlterRoleSetStmt);
+	const char *currentDatabaseName = CurrentDatabaseName();
+	List *commandList = NIL;
+	bool isnull = false;
+
+	const char *databaseName =
+		GetDatabaseNameFromDbRoleSetting(tuple, DbRoleSettingDescription);
+
+	/*
+	 * session defaults for databases other than the current one are skipped
+	 */
+	if (databaseName != NULL &&
+		pg_strcasecmp(databaseName, currentDatabaseName) != 0)
+	{
+		ereport(NOTICE, (errmsg("Citus partially supports ALTER ROLE .. IN DATABASE"
+								" .. SET  for distributed databases"),
+						 errdetail("Citus propagates the session defaults that affect "
+								   "current database"),
+						 errhint("You can manually change attributes of roles "
+								 "on workers")));
+		return NULL;
+	}
+
+	if (databaseName != NULL)
+	{
+		stmt->database = pstrdup(databaseName);
+	}
+
+	const char *roleName = GetRoleNameFromDbRoleSetting(tuple, DbRoleSettingDescription);
+
+	/*
+	 * default roles are skipped, because reserved roles
+	 * cannot be altered.
+	 */
+	if (roleName != NULL && IsReservedName(roleName))
+	{
+		return NULL;
+	}
+
+	if (roleName != NULL)
+	{
+		stmt->role = makeNode(RoleSpec);
+		stmt->role->location = -1;
+		stmt->role->roletype = ROLESPEC_CSTRING;
+		stmt->role->rolename = pstrdup(roleName);
+	}
+
+	Datum setconfig = heap_getattr(tuple, Anum_pg_db_role_setting_setconfig,
+								   DbRoleSettingDescription, &isnull);
+	ArrayType *array = DatumGetArrayTypeP(setconfig);
+	int i;
+
+	for (i = 1; i <= ARR_DIMS(array)[0]; i++)
+	{
+		Datum d = array_ref(array, 1, &i,
```

Comment:
In all other similar situations we used `deconstruct_array`. Maybe do the same here.

Reply:
did this in https://github.com/citusdata/citus/pull/3510/commits/e90a4adcb4bb951216ed568d8e7f7a092cd9e4d6.

Thanks for the advice